aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -3789,7 +3790,49 @@ public SignInProviderConfig(final Class<? extends SignInProvider> signInProvider
     }
 }
 
-class AWSMobileClientStore {
+interface KeyValueStore {
```

Comment:
Could this be in a new file?  I know the previous pattern here seems to be to throw everything into this one massive [4060 line file](https://github.com/aws-amplify/aws-sdk-android/blob/ac38a5b18fb546c54251279e07b24d84c70f3ece/aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java#L4060), but perhaps we could break the trend now? :)

Reply:
haha yea, we can do that